diesel_tests/tests/combination.rs

```diff
@@ -117,6 +117,65 @@ fn except() {
     assert_eq!(expected_data, data);
 }
 
+#[test]
+fn union_with_limit() {
+    use crate::schema::users::dsl::*;
+
+    let conn = &mut connection();
+    let data = vec![
+        NewUser::new("Sean", None),
+        NewUser::new("Tess", None),
+        NewUser::new("Jim", None),
+    ];
+    insert_into(users).values(&data).execute(conn).unwrap();
+    let data = users.order(id).load::<User>(conn).unwrap();
+    let _sean = &data[0];
+    let tess = &data[1];
+    let _jim = &data[2];
+
+    let data: Vec<User> = users
+        .filter(id.le(tess.id))
+        .union(users.filter(id.ge(tess.id)))
+        .limit(2)
+        .load(conn)
+        .unwrap();
+
+    // Cannot determine any `expected_data` to match against because the database might return
+    // different rows
+    // The positional_order_by fix is not applicable, because ORDER should appear before LIMIT
```

Comment:
Could you elaborate why this is not possible?

Reply:
I've added a more extensive comment, tell me if that's enough or if you need more

Reply:
The explanation sounds fine to me, thanks for adding it. :+1:

That written: This indicates that we likely should change how `positional_order_by` is applied to the type. It likely should behave similar to the newly implemented functions instead of being stacked on top of the existing query. 

(That's something for a followup PR)